src/app/json.cpp

```diff
@@ -213,8 +213,10 @@ static void from_json(const Json &aJson, Config &aConfig)
 
     if (aJson.contains("LogFile"))
     {
-        auto logger     = std::make_shared<FileLogger>(aJson["LogFile"], logLevel);
-        aConfig.mLogger = std::dynamic_pointer_cast<Logger>(logger);
+        std::shared_ptr<FileLogger> logger;
+
+        SuccessOrThrow(FileLogger::Create(logger, aJson["LogFile"], logLevel));
```

Comment:
`SuccessOrThrow` should only be called for json related code?

Reply:
Yes, will refactor this.

Reply:
Fine to use exception only in this file.